src/sagemaker/serve/builder/model_builder.py

```diff
@@ -1157,6 +1161,15 @@ def optimize(
             Model: A deployable ``Model`` object.
         """
 
+        # TODO: ideally these dictionaries need to be sagemaker_core shapes
```

Comment:
TODO: Ideally validations should happen at the start of the flow, not after everything is set up in the wrapper. Lets fast follow this and gradually shift the validations to helper functions in the outer scope here